pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -382,6 +383,43 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 	// Render the desired objects from the CRD and create or update them.
 	reqLogger.V(3).Info("rendering components")
 
+	var authenticationCR *operatorv1.Authentication
+	// Fetch the Authentication spec. If present, we use to configure user authentication.
+	authenticationCR, err = utils.GetAuthentication(ctx, r.client)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error while fetching Authentication", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// set auth type in the config
+	authType := ""
+
+	if authenticationCR != nil && authenticationCR.Status.State != operatorv1.TigeraStatusReady {
+		r.status.SetDegraded(operatorv1.ResourceNotReady,
+			fmt.Sprintf("Authentication is not ready authenticationCR status: %s", authenticationCR.Status.State),
+			nil, reqLogger)
+		return reconcile.Result{}, nil
+	} else if authenticationCR != nil && !utils.IsDexDisabled(authenticationCR) {
+		// Do not include DEX TLS Secret Name if authentication CR does not have type Dex
+		secret := render.DexTLSSecretName
+		certificate, err := certificateManager.GetCertificate(r.client, secret, helper.TruthNamespace())
+		if err != nil {
+			r.status.SetDegraded(operatorv1.CertificateError, fmt.Sprintf("Failed to retrieve %s", secret),
+				err, reqLogger)
+			return reconcile.Result{}, err
+		} else if certificate == nil {
+			reqLogger.Info(fmt.Sprintf("Waiting for secret '%s' to become available", secret))
+			r.status.SetDegraded(operatorv1.ResourceNotReady,
+				fmt.Sprintf("Waiting for secret '%s' to become available", secret),
+				nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		trustedBundle.AddCertificates(certificate)
```

Comment:
Theoretically speaking, trustedBundle can be nil here. I'd suggest creating the bundle after this line:

```
if installationSpec.Variant == operatorv1.TigeraSecureEnterprise {
trustedBundle = certificateManager.CreateTrustedBundle()
```

Then change the following line:

```
- trustedBundle = certificateManager.CreateTrustedBundle()
+ trustedBundle.AddCertificates(prometheusCertificate)
```

Reply:
thanks for catching this :+1: